canals/pipeline/pipeline.py

```diff
@@ -152,8 +153,18 @@ def from_dict(cls: Type[T], data: Dict[str, Any], **kwargs) -> T:
             else:
                 if "type" not in component_data:
                     raise PipelineError(f"Missing 'type' in component '{name}'")
+
                 if component_data["type"] not in component.registry:
-                    raise PipelineError(f"Component '{component_data['type']}' not imported.")
+                    try:
+                        # Import the module first...
+                        module, _ = component_data["type"].rsplit(".", 1)
+                        importlib.import_module(module)
+                        # ...then try again
+                        if component_data["type"] not in component.registry:
+                            raise PipelineError()
+                    except (ImportError, PipelineError) as e:
+                        raise PipelineError(f"Component '{component_data['type']}' not imported.") from e
```

Comment:
This is the core logic: in `from_dict` we try to import the module containing the type of the components listed in the dictionary.